stan/math/rev/fun/quad_form.hpp

```diff
@@ -20,13 +20,13 @@ class quad_form_vari_alloc : public chainable_alloc {
   inline void compute(const Eigen::Matrix<double, Ra, Ca>& A,
                       const Eigen::Matrix<double, Rb, Cb>& B) {
     matrix_d Cd = B.transpose() * A * B;
+    if (sym_) {
+      matrix_d M = 0.5 * (Cd + Cd.transpose());
+      Cd = M;
+    }
     for (int j = 0; j < C_.cols(); j++) {
       for (int i = 0; i < C_.rows(); i++) {
-        if (sym_) {
-          C_(i, j) = var(new vari(0.5 * (Cd(i, j) + Cd(j, i)), false));
-        } else {
-          C_(i, j) = var(new vari(Cd(i, j), false));
-        }
+        C_(i, j) = var(new vari(Cd(i, j), false));
```

Comment:
Why does this work and the previous thing didn't? They look equivalent to me.

Reply:
This part of the code is a refactor; there should be nothing different there other than removing an if statement within the loops. 

The part that’s different is in construction of the `Cd` matrix. There was a flag that was unused. Instead of adding a new vari subclass, I just used the one that was already written and it matches the prim implementation.